rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -144,6 +149,47 @@ impl CanisterHttpPoolManagerImpl {
             .collect()
     }
 
+    fn generate_api_boundary_node_addrs(&self) -> Vec<String> {
```

Comment:
nit: you are not really generating the BN addrs. Maybe get_socks_proxy_addrs is better.

Also up to you if you want to start including the addresses in this PR. Namely, I am fine if you don't have the `SOCKS_PROXY_DL_ACTIVE` in this file. Or ideally we have a single flag that enables or disables the features.

So we can either have the flag only here and do the socks request if the addrs in the RPC is non empty. Or we always add the list of BNs here and you keep the adapter code as is. Your choice.

However, not sure it makes sense to have two places where we enable and disable the feature.